src/metadata.ts

```diff
@@ -21,7 +21,7 @@ export function normalizeMetadata(input?: File | Response | BufferLike | StreamL
   }
   if (input instanceof Response) {
     const contentDisposition = input.headers.get("content-disposition")
-    const filename = contentDisposition && contentDisposition.match(/;\s*filename\*?=["']?(.*?)["']?$/i)
+    const filename = contentDisposition && contentDisposition.match(/filename=['"]?([^'";]+)['"]?(?:;|$)/i)   
```

Comment:
The previous regex did not properly interpret the `*=` syntax (we should not match the encoding identifier that follows it), but [it's a real thing](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition) — even recommended — so we should keep matching it.

Reply:
PS: I don't think we should bother with the general syntax, only the case of UTF-8 because it is common and it's the only one we can decode in the browser without extra libraries.

If you don't mind adding that feature correctly while we're at it, we should be able to match **and decode** the filename "όνομα αρχείου.txt" in `Content-Disposition: attachment; filename* = UTF-8''%CF%8C%CE%BD%CE%BF%CE%BC%CE%B1%20%CE%B1%CF%81%CF%87%CE%B5%CE%AF%CE%BF%CF%85.txt` (I picked the first non-latin alphabet that came to mind, feel free to use another example). And, if I read the grammar correctly, there is optional whitespace between `*` and `=` as well as between `=` and the value, so I added them in my example (but not between `filename` and `*`, which is not allowed by the grammar).

Reply:
Indeed, my updates was missing something  from the previous implementation, my bad.  
That should work with my last update bda1d4ed41f0ffc48ffa19220299d5204d2fe18c